tests/e2e/tests/test_ingress.py

```diff
@@ -0,0 +1,175 @@
+#
+# Copyright 2023 Canonical, Ltd.
+#
+import json
+import logging
+import subprocess
+from pathlib import Path
+
+import pytest
+from e2e_util import config, harness, util
+from e2e_util.config import MANIFESTS_DIR
+
+LOG = logging.getLogger(__name__)
+
+
+def check_pvc_bound(p: subprocess.CompletedProcess) -> bool:
+    out = json.loads(p.stdout.decode())
+    for pvc in out["items"]:
+        if pvc["metadata"]["name"] == "myclaim" and pvc["status"]["phase"] == "Bound":
+            return True
+    return False
+
+
+def test_ingress(h: harness.Harness, tmp_path: Path):
+    if not config.SNAP:
+        pytest.fail("Set TEST_SNAP to the path where the snap is")
+
+    snap_path = (tmp_path / "k8s.snap").as_posix()
+
+    LOG.info("Create instance")
+    instance_id = h.new_instance()
+
+    util.setup_k8s_snap(h, instance_id, snap_path)
+    h.exec(instance_id, ["k8s", "init"])
+    util.setup_network(h, instance_id)
+
+    out = h.exec(
+        instance_id,
+        ["k8s", "enable", "ingress"],
+        capture_output=True,
+    )
+    assert "enabled" in out.stderr.decode()
```

Comment:
note: this will break after https://github.com/canonical/k8s-snap/pull/62 as everything is not printed to `stderr`

Reply:
Adjusted the tests, I've also spotted a missing change and fixed it. The PR didn't seem to update the `stderr` in `setup_network` function.